sdk/clientcore/core/src/main/java/io/clientcore/core/http/pipeline/BearerTokenAuthenticationPolicy.java

```diff
@@ -129,4 +111,24 @@ public Response<?> process(HttpRequest httpRequest, HttpPipelineNextPolicy next)
         }
         return httpResponse;
     }
+
+    /**
+     * Authorizes the request with the bearer token acquired using the specified {@code tokenRequestContext}
+     *
+     * @param context the HTTP pipeline context.
+     * @param tokenRequestContext the token request context to be used for token acquisition.
+     */
+    public void setAuthorizationHeader(HttpRequest context, TokenRequestContext tokenRequestContext) {
```

Comment:
Any reason this is public?

Reply:
Yeah, this allows downstream classes to supply their own bearer token from their respective cache and set it.

Reply:
Could be protected though?